### PR TITLE
[wip] support RANGE frame for window functions

### PR DESCRIPTION
**Pull request by akuzm**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by alesapin** [comment]
Failure in test keeper https://github.com/ClickHouse/ClickHouse/issues/18950.

**Comment by akuzm** [comment]
Fuzzer https://github.com/ClickHouse/ClickHouse/issues/19579

**Comment by akuzm** [comment]
Integration test failures are not related to the PR.

File: src/Processors/Transforms/WindowTransform.cpp [hunk 1]
```diff
@@ -156,34 +484,158 @@ void WindowTransform::transform(Chunk & chunk)
...
+        assert(first_not_ready_row == blocksEnd());
+
+        // FIXME do we really have to do this?
+        output.finish();
+
+        return Status::Finished;
+    }
+
+    // Consume input data if we have any ready.
+    if (!has_input && input.hasData())
```
Comment (Alex-Cheng) [inline src/Processors/Transforms/WindowTransform.cpp:555]: Is the variable `has_input` required? I see the `input.pullData(true)` has set the input as 'not needed' and the status is Ready. Is it any possibility that another input data block comes before `work()` executes? There is a queue `blocks` for input data blocks in `work()`.

**Pull request merged**